src/Integration.Vsix.UnitTests/InfoBar/InfoBarManagerTests.cs

```diff
@@ -303,6 +309,60 @@ public void AttachInfoBarWithButtons_ButtonClickedEvent_RaisedWithTheRightButton
             eventHandler.VerifyNoOtherCalls();
         }
 
+        #region MainWindow tests
+        // The ToolWindow tests above cover the common implementation of gold bar functionality.
+        // These MainWindow tests just cover a few extra main window specific cases.
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_ArgChecks()
+        {
+            var testSubject = CreateTestSubject();
+
+            Action act = () => testSubject.AttachInfoBarToMainWindow(null, SonarLintImageMoniker.OfficialSonarLintMoniker);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("message");
+
+            act = () => testSubject.AttachInfoBarToMainWindow("a message", SonarLintImageMoniker.OfficialSonarLintMoniker, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("buttonTexts");
+        }
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_AttachAndDetach()
+        {
+            var sp = new ConfigurableServiceProvider();
```

Comment:
This test class is written in the "legacy" style i.e. class-instance level variables and a `TestInitialize` method. I wanted to make the new tests independent so they don't rely on the `TestInitialize. There is already an instance-level variable called `serviceProvider` so I used `sp` here.

Reply:
Makes sense, I think I had to do the same with "shell" -> shellMock somewhere here